### PR TITLE
Submit Profile Updates

### LINKED ISSUE #1135: refactor CMakeLists.txt in game of life example
The file CMakeLists.txt in game of life needs a style refactoring. Cmake commands are written mixed with lower and upper case letters.

We should use the some style like in PIConGPU.

### PR DESCRIPTION
**Pull request by ax3l**
Close #1135, updates the GoL `CMakeLists.txt` with PIConGPU-style groups and modernized commands, dependencies, warnings, includes, `sm` > 2.0, dependency version requirements and scripts.

### To Do

- [x] inline comments
- [x] include `make_directories` change from #1123
- [x] do compile & install & run test again at the end

### CODE REVIEW

**Comment by ax3l** [comment]
@PrometheusPi if you don't mind, I would like to have @psychocoderHPC a view on it since he has more experience with templates and knows the machines.

File: src/picongpu/submit/lawrencium-lbnl/picongpu.profile.example [hunk 1]
```diff
@@ -42,8 +44,12 @@ export PATH=$HOME/bin:$PATH
 export PATH=$PATH:$PICSRC/src/splash2txt/build
 export PATH=$PATH:$PICSRC/src/tools/bin
 
-# "tbg" default options:
+# send me a mails on job (b)egin, (e)nd, (a)bortion or (n)o mails
```
Comment (PrometheusPi) [inline src/picongpu/submit/lawrencium-lbnl/picongpu.profile.example]: This is a slurm system, where these option do not exist.

File: src/picongpu/submit/titan-ornl/batch_profile.tpl [hunk 2]
```diff
@@ -18,12 +18,10 @@
...
 #
 
-
-
 ## calculations will be performed by tbg ##
 TBG_queue="batch"
-TBG_mailSettings="bea"
-TBG_mailAdress="someone@example.com"
+TBG_mailAddress=${MY_MAIL:-"someone@example.com"}
+TBG_mailSettings=${MY_MAILNOTIFY:-"n"}
```
Comment (PrometheusPi) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: This is a batch job. Do we need sending mails?

File: src/picongpu/submit/lawrencium-lbnl/picongpu.profile.example (hunk 1 above)
Comment (ax3l) [inline src/picongpu/submit/lawrencium-lbnl/picongpu.profile.example]: you're right, it is!

File: src/picongpu/submit/titan-ornl/batch_scorep_profile.tpl [hunk 3]
```diff
@@ -38,7 +36,7 @@ TBG_nodes=!TBG_tasks
 #PBS -N !TBG_jobName
 #PBS -l nodes=!TBG_nodes
 # send me a mail on (b)egin, (e)nd, (a)bortion
```
Comment (psychocoderHPC) [inline src/picongpu/submit/titan-ornl/batch_scorep_profile.tpl]: should we remove this documentation at this point?

File: src/picongpu/submit/titan-ornl/batch_profile.tpl [hunk 4]
```diff
@@ -38,7 +36,7 @@ TBG_nodes=!TBG_tasks
 #PBS -N !TBG_jobName
 #PBS -l nodes=!TBG_nodes
 # send me a mail on (b)egin, (e)nd, (a)bortion
```
Comment (psychocoderHPC) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: should we remove this documentation at this point?

Comment (ax3l) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: but it's pretty useful when reading the final `submit.start` file

File: src/picongpu/submit/titan-ornl/batch_scorep_profile.tpl (hunk 3 above)
Comment (ax3l) [inline src/picongpu/submit/titan-ornl/batch_scorep_profile.tpl]: see above

File: src/picongpu/submit/titan-ornl/batch_profile.tpl (hunk 4 above)
Comment (ax3l) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: but we could change those to: `send me a mail on (b)egin, (e)nd, (a)bortion or (n)one`

Comment (PrometheusPi) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: Would it be possible to replace `me` by the value of `$TBG_mailAddress`?

Comment (ax3l) [inline src/picongpu/submit/titan-ornl/batch_profile.tpl]: yes that works.
or just skipping the "me" because the mail is written the line below

**Pull request merged**